src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,4 +60,40 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
```

Comment:
Method is complicated to comprehend at first. Would help if there are comments for each if block statements rather than just the first if block statement. 

Can consider using a more meaningful variable than `i` for example `currLength`